android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/UploadManager.java

```diff
@@ -26,76 +30,257 @@
+                                                 @NonNull UploadSession uploadSession,
+                                                 @NonNull UploadValidationStatus uploadValidationStatus) {
+        checkNotNull(uploadFile, "uploadFile cannot be null");
+        checkNotNull(uploadSession, "uploadSession cannot be null");
+        checkNotNull(uploadValidationStatus, "uploadValidationSession cannot be null");
+
+        switch (uploadValidationStatus.getStatus()) {
+            case REQUESTED:
+                return uploadToS3(uploadFile, uploadSession);
+            case SUCCEEDED:
+            case DUPLICATE:
+                return dequeueUpload(uploadFile.filename);
+            case VALIDATION_IN_PROGRESS:
+                LOG.debug("Validation in progress for filename: " + uploadFile.filename +
+                        ", uploadId" + uploadSession.getId());
+                break;
+            case VALIDATION_FAILED:
+                LOG.debug("Validation failed for filename: " + uploadFile.filename +
+                        ", uploadId" + uploadSession.getId());
+                break;
+            case UNKNOWN:
+            default:
+                LOG.warn("Unknown status for uploadId: " + uploadValidationStatus.getId());
+        }
+        // return a successful completion when there's nothing to do
+        // leaves the files queued, in case there's future recovery options (e.g. next
+        // release), will add removal for permanent failures
+        return Completable.complete();
+    }
 
+    /**
+     * Deletes a file from disk and removes it from the upload queue.
+     *
+     * @param filename the file's name
+     * @return dequeueing completable
+     */
+    @NonNull
+    Completable dequeueUpload(@NonNull String filename) {
+        checkNotNull(filename, "filename required");
+
+        if (getFile(filename).delete()) {
+            LOG.warn("Successfully deleted upload file: " + filename + ", "
+                    + "removing upload from queue");
+            return Completable.fromAction(() -> uploadDAO.removeUploadAndSession(filename));
+        } else {
+            LOG.warn("Failed to delete upload file: " + filename);
+            return Completable.error(new IOException("Failed to delete upload file: " + filename));
+        }
+    }
+
+    /**
+     * @param uploadId Bridge upload identifier
+     * @return upload validation status single
+     */
+    @NonNull
+    Single<UploadValidationStatus> getUploadValidationStatus(@NonNull String uploadId) {
+        checkNotNull(uploadId, "uploadId required");
+
+        return RxUtils.toBodySingle(api.getUploadStatus(uploadId));
+    }
+
+    @NonNull
+    Completable uploadToS3(UploadFile uploadFile, UploadSession session) {
         File file = getFile(uploadFile.filename);
+        checkArgument(file.exists(), "Non-existent file: " + file.getAbsolutePath());
+
+        Single<UploadSession> sessionSingle = Single.just(session)
```

Comment:
What is going on here?  You take an UploadSession, and wrap it in a Single and then operate on that same variable?